pipeline_tools/create_envelope.py

```diff
@@ -1,277 +1,361 @@
+    """Query the Ingest service to get the url for creating envelopes.
 
     Args:
-        submit_url (str): the root url of the ingest service
-        auth_headers (dict): headers to use for auth
-        http_requests (HttpRequests): the HttpRequests object to use
+        submit_url (str): The root url of the Ingest service for submissions.
+        auth_headers (dict): Dict representing headers to use for auth.
+        http_requests (http_requests.HttpRequests): The HttpRequests object to use for talking to Ingest.
 
     Returns:
-        envelope_url (str): A string giving the url for creating envelopes
+        envelope_url (str): A string giving the url for creating envelopes.
 
     Raises:
-        requests.HTTPError: for 4xx errors or 5xx errors beyond timeout
+        requests.HTTPError: For 4xx errors or 5xx errors beyond timeout.
     """
     print('Getting envelope url from {}'.format(submit_url))
-    response = http_requests.get(submit_url,
-                                 headers=auth_headers)
-    envelope_url = get_subject_url(response.json(), 'submissionEnvelopes')
+    response = http_requests.get(submit_url, headers=auth_headers)
+    envelope_url = get_subject_url(target_dic=response.json(), subject='submissionEnvelopes')
     return envelope_url
 
 
 def create_submission_envelope(envelope_url, auth_headers, http_requests):
-    """Creates a new submission envelope
+    """Creates a new submission envelope and return its content.
 
     Args:
-        envelope_url (str): the url for creating envelopes
-        auth_headers (dict): headers to use for auth
-        http_requests (HttpRequests): HttpRequests object to use
+        envelope_url (str): The url for creating envelopes.
+        auth_headers (dict): Dict representing headers to use for auth.
+        http_requests (http_requests.HttpRequests): The HttpRequests object to use for talking to Ingest.
 
     Returns:
-        envelope_js (dict): Dict representing the JSON response to the request
+        envelope_dict (dict): Dict representing the JSON response to the request of creating the envelope.
 
     Raises:
-        requests.HTTPError: for 4xx errors or 5xx errors beyond timeout
+        requests.HTTPError: For 4xx errors or 5xx errors beyond timeout.
     """
     print('Creating submission envelope at {0}'.format(envelope_url))
-    response = http_requests.post(envelope_url,
-                                  '{}',
-                                  headers=auth_headers)
-    envelope_js = response.json()
-    return envelope_js
+    response = http_requests.post(envelope_url, '{}', headers=auth_headers)
+    envelope_dict = response.json()
+    return envelope_dict
 
 
-def get_analysis_process(analyses_url, auth_headers, analysis_workflow_id, http_requests):
-    """Checks the submission envelope for an analysis process with a protocol_id
-    that matches the analysis workflow id.
+# FIXME: questionable in new bundle structure :(
```

Comment:
What does this mean?

Reply:
This is a to-do placeholder for me 😄 Since we are separating analysis_json into analysis_protocol and analysis_process now, we have to make an additional call like `curl -X PUT -H “Content-Type: text/uri-list” -d “http://api.ingest.integration.data.humancellatlas.org/protocols/xxx” “http://api.ingest.integration.data.humancellatlas.org/processes/xxx/protocols”`

And I'm still working on that.